service/history/transferQueueActiveTaskExecutor.go

```diff
@@ -1434,3 +1439,18 @@ func (t *transferQueueActiveTaskExecutor) applyParentClosePolicy(
 		}
 	}
 }
+
+func copyChildWorkflowInfos(
```

Comment:
Can we have these as helper functions in a common helper.

Reply:
i am going to cherry-pick this PR, so let me or alex follow up with some dedicated PRs to move things to their right place.